### PR TITLE
Add string support for intrinsic `max`

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: https://github.com/lfortran/lfortran/issues/2841

### CODE REVIEW

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -3017,21 +3028,40 @@ namespace Max {
...
 
         ASR::expr_t* test;
         body.push_back(al, b.Assignment(result, args[0]));
-        for (size_t i = 1; i < args.size(); i++) {
-            test = make_Compare(make_IntegerCompare_t, args[i], Gt, result);
-            Vec<ASR::stmt_t *> if_body; if_body.reserve(al, 1);
-            if_body.push_back(al, b.Assignment(result, args[i]));
-            body.push_back(al, STMT(ASR::make_If_t(al, loc, test,
-                if_body.p, if_body.n, nullptr, 0)));
+        if (return_type->type == ASR::ttypeType::Integer) {
```
Comment (czgdp1807) [inline src/libasr/pass/intrinsic_function_registry.h:3039]:
```suggestion
        if (ASR::is_a<ASR::Integer_t>(*return_type)) {
```

File: src/libasr/pass/intrinsic_function_registry.h [hunk 2]
```diff
@@ -3017,21 +3028,40 @@ namespace Max {
...
+            for (size_t i = 1; i < args.size(); i++) {
+                test = make_Compare(make_IntegerCompare_t, args[i], Gt, result);
+                Vec<ASR::stmt_t *> if_body; if_body.reserve(al, 1);
+                if_body.push_back(al, b.Assignment(result, args[i]));
+                body.push_back(al, STMT(ASR::make_If_t(al, loc, test,
+                    if_body.p, if_body.n, nullptr, 0)));
+            }
+        } else if (return_type->type == ASR::ttypeType::Real) {
```
Comment (czgdp1807) [inline src/libasr/pass/intrinsic_function_registry.h:3047]:
```suggestion
        } else if (ASR::is_a<ASR::Real_t>(*return_type)) {
```

File: src/libasr/pass/intrinsic_function_registry.h [hunk 3]
```diff
@@ -3017,21 +3028,40 @@ namespace Max {
...
+            for (size_t i = 1; i < args.size(); i++) {
+                test = make_Compare(make_RealCompare_t, args[i], Gt, result);
+                Vec<ASR::stmt_t *> if_body; if_body.reserve(al, 1);
+                if_body.push_back(al, b.Assignment(result, args[i]));
+                body.push_back(al, STMT(ASR::make_If_t(al, loc, test,
+                    if_body.p, if_body.n, nullptr, 0)));
+            }
+        } else if (return_type->type == ASR::ttypeType::Character) {
```
Comment (czgdp1807) [inline src/libasr/pass/intrinsic_function_registry.h:3055]:
```suggestion
        } else if (ASR::is_a<ASR::Character_t>(*return_type)) {
```

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: tests/reference/asr-fn5-3d75eb7.stderr
Comment (czgdp1807) [inline tests/reference/asr-fn5-3d75eb7.stderr:1]: This error file got created. Shouldn't have happened. Needs to be debugged.

File: tests/reference/asr-string_19-d880475.stderr
Comment (czgdp1807) [inline tests/reference/asr-string_19-d880475.stderr:1]: File with same error.

**Review by czgdp1807 — DISMISSED** [review]

File: tests/reference/asr-fn5-3d75eb7.stderr
Comment (czgdp1807) [reply tests/reference/asr-fn5-3d75eb7.stderr:1]: CI for FPM also fails with this error.

**Review by certik — APPROVED** [review]
I think this looks good.

**Pull request merged**